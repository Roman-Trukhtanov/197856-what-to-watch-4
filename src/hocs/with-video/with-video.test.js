import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withVideo from "./with-video.js";

const mockMovieData = {
  id: 1,
  title: `Snatch`,
  genre: `Comedy`,
  bgColor: `#FDFDFC`,
  coverSrc: `https://some-link`,
  bigPosterSrc: `https://some-link`,
  previewImgSrc: `https://some-link`,
  isFavorite: false,
  fullVideo: {
    className: `player__video`,
    src: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    type: `video/mp4`,
    isAutoPlay: true,
    isLoop: false,
    isMute: false,
  },
  previewVideo: {
    src: `https://some-link`,
    type: `video/mp4`,
    isAutoPlay: true,
    isLoop: true,
    isMute: true,
  },
  details: {
    rate: 3,
    releaseYear: 2000,
    ratingCount: 100,
    level: `Bad`,
    description: `Description`,
    director: `Guy Ritchie`,
    runTime: 104,
    starring: [`Jason Statham`, `Brad Pitt`, `Benicio Del Toro`],
  },
};

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withVideo(MockComponent);

it(`withAudio is rendered correctly`, () => {
  const {previewVideo, previewImgSrc} = mockMovieData;

  const tree = renderer.create((
    <MockComponentWrapped
      isStartPlaying={true}
      previewImgSrc={previewImgSrc}
      videoData={previewVideo}
    />
  ), {
    createNodeMock() {
      return {
        addEventListener() {},
        play() {},
      };
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
