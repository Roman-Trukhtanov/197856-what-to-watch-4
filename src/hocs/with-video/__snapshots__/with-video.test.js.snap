// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAudio is rendered correctly 1`] = `
<div>
  <video
    className=""
    controls={false}
    loop={true}
    playsInline={true}
    poster="https://some-link"
    preload="auto"
    style={
      Object {
        "height": "100%",
        "objectFit": "cover",
        "width": "100%",
      }
    }
  >
    <source
      src="https://some-link"
      type="video/mp4"
    />
  </video>
</div>
`;
